src/kixi/hecuba/onyx/jobs/weather_fetcher.clj

```diff
@@ -25,6 +33,23 @@
         body (vec (rest data))]
     (map #(zipmap headers %) body)))
 
+;; The McKiver method (or British Gas method) as employed by the Met Office
+;; For more info see http://www.vesma.com/ddd/ddcalcs.htm
+(defn calc-degreedays-mckiver [tbase tmin tmax]
+  (double
+   (cond
+     (> tmin tbase)                0.0
+     (> (/ (+ tmax tmin) 2) tbase) (/ (- tbase tmin) 4)
+     (>= tmax tbase)               (- (/ (- tbase tmin) 2) (/ (- tmax tbase) 4))
+     (< tmax tbase)                (- tbase (/ (+ tmax tmin) 2))
+     :else -1)))
+
+(defn get-max-and-min [daily-readings]
+  (try
+    (let [data (map (fn [t] (Float/parseFloat (:value t))) daily-readings)]
+      {:max (apply max data) :min (apply min data)})
+    (catch Exception e (str "Exception caught in get-max-and-min:" (.getMessage e)))))
```

Comment:
Return type changed on exception. Healthy gets a map, exception gets a string. This tends to cause another exception which masks what's actually wrong. Better to throw, or log and return a default/nil. Probably throw here i imagine.